packages/repo-tools/src/commands/api-reports/api-reports.ts

```diff
@@ -15,44 +15,51 @@
  */
 
 import { OptionValues } from 'commander';
-import { resolve as resolvePath } from 'path';
 import fs from 'fs-extra';
-import { spawnSync } from 'child_process';
 import {
-  findSpecificPackageDirs,
   createTemporaryTsConfig,
-  findPackageDirs,
   categorizePackageDirs,
   runApiExtraction,
   runCliExtraction,
   buildDocs,
 } from './api-extractor';
+import { findPackageDirs, paths as cliPaths } from '../../lib/paths';
+import { generateTSC } from './generateTSC';
 
-export default async (paths: string[], opts: OptionValues) => {
-  const tmpDir = resolvePath(
-    process.cwd(),
+type Options = {
+  ci?: boolean;
+  docs?: boolean;
+  tsc?: boolean;
+  paths?: string[];
+  allowWarnings?: string[] | boolean;
+  omitMessages?: string[];
+} & OptionValues;
+
+export const buildApiReports = async (opts: Options) => {
+  const tmpDir = cliPaths.resolveTargetRoot(
     './node_modules/.cache/api-extractor',
   );
-  const projectRoot = resolvePath(process.cwd());
+
   const isCiBuild = opts.ci;
   const isDocsBuild = opts.docs;
   const runTsc = opts.tsc;
 
-  const selectedPackageDirs = await findSpecificPackageDirs(paths);
+  const parsedPaths = parseArrayOption(opts.paths);
+  const isAllPackages = !Array.isArray(parsedPaths) || !parsedPaths?.length;
+  const selectedPaths = isAllPackages ? await getWorkspacePkgs() : parsedPaths;
+  const selectedPackageDirs = await findPackageDirs(selectedPaths);
 
-  if (selectedPackageDirs && isCiBuild) {
-    throw new Error(
-      'Package path arguments are not supported together with the --ci flag',
-    );
-  }
-  if (!selectedPackageDirs && !isCiBuild && !isDocsBuild) {
+  const allowWarnings = parseArrayOption(opts.allowWarnings);
+  const omitMessages = parseArrayOption(opts.omitMessages);
+
+  if (isAllPackages && !isCiBuild && !isDocsBuild) {
     console.log('');
     console.log(
       'TIP: You can generate api-reports for select packages by passing package paths:',
     );
     console.log('');
     console.log(
-      '       yarn build:api-reports packages/config packages/core-plugin-api',
+      '       yarn build:api-reports -p packages/config -p packages/core-plugin-api,plugins/*',
```

Comment:
I'm kinda tempted to keep this as is tbh, i.e. have the args be the paths, rather than put them behind the `--path` flag. What's your thinking around adding the flag?

Reply:
I was looking at [this](https://github.com/tj/commander.js/blob/master/docs/options-taking-varying-arguments.md#parsing-ambiguity) and this was one of the options to solve the parsing ambiguity and thought that having it as an option forces somehow how to interpret the command

We can totally keep it as is. It's a limitation to leave the `Arguments` first and `Options` last. in that case or use `--`

```bash
# DOESN'T WORKS path1 is considered path, but path2 will be treated as omitMessages
api-reports path1 --omit-messages  message1,message2 path2

# WORKS pass all paths first
api-reports path1 path2 --omit-messages  message1,message2

# WORKS pass more paths after --  
api-reports path1  --omit-messages  message1,message2 -- path2
```

Reply:
Perhaps two separate options `--allow-warnings` and `--allow-warnings-for-paths <paths>`?

Reply:
Ok `--allow-warnings <paths>` and `--allow-warnings-all` looks good?

Reply:
updated  with `--allow-warnings <paths>` and `--allow-warnings-all` for now

Reply:
Perhaps `--allow-all-warnings` to make it read better? 😅